frontend/src/Pages/Programs.tsx

```diff
@@ -0,0 +1,101 @@
+import { useAuth } from '@/useAuth';
+import { useState, useRef } from 'react';
+import ProgramCard from '@/Components/ProgramCard';
+import SearchBar from '@/Components/inputs/SearchBar';
+import { Program, ServerResponse, ViewType, UserRole } from '@/common';
+import useSWR from 'swr';
+import DropdownControl from '@/Components/inputs/DropdownControl';
+import { AxiosError } from 'axios';
+import { PlusCircleIcon } from '@heroicons/react/24/outline';
+import ToggleView from '@/Components/ToggleView';
+
+export default function Programs() {
+    const { user } = useAuth();
+    const addProgramModal = useRef<HTMLDialogElement>(null);
+
+    if (!user) {
+        return null;
+    }
+
+    const [activeView, setActiveView] = useState<ViewType>(ViewType.Grid);
+    const [searchTerm, setSearchTerm] = useState('');
+    const [order, setOrder] = useState('asc');
+    const { data, error, mutate } = useSWR<ServerResponse<Program>, AxiosError>(
```

Comment:
I'm not sure if I'm interpreting this correctly, but if you are asking if instead of using useSWR we should use the routeLoader, my answer is its not necessary right now. Preston and I are going to decide if we're going to fully implement react-router or switch to a different library, so for now the use of useSWR is okay. If that didn't answer the question let me know!

Reply:
I think we can probably handle that when we decide whatever our path is after evaluating our use of `useSWR` / react router